src/services/weatherService.js

```diff
@@ -29,6 +29,26 @@ class WeatherService {
       throw error;
     }
   }
+
+  async getWeeklyForecast(latitude, longitude) {
+    try {
+      const params = {
+        latitude,
+        longitude,
+        daily:
+          'weather_code,temperature_2m_max,temperature_2m_min,precipitation_sum,wind_speed_10m_max',
+      };
+
+      const response = await axios.get(`${baseUrl}/forecast`, { params });
+
+      return {
+        daily: response.data.daily,
+      };
+    } catch (error) {
+      console.log('Error fetching weekly forecast:', error.message);
+      throw error;
```

Comment:
Please remove `throw error`, it makes code execution stop here, and `hasError` state will never be `true`
![image](https://github.com/user-attachments/assets/53fb4ba6-b428-4103-a0d9-68bfac304e1d)